addon/components/g-map.js

```diff
@@ -29,18 +29,20 @@ export default Ember.Component.extend({
   }),
 
   didInsertElement() {
-    this._super();
-    if (isEmpty(this.get('map')) &&
-      (typeof FastBoot === 'undefined')) {
-      const canvas = this.$().find('.g-map-canvas').get(0);
-      const options = this.get('permittedOptions');
-      this.set('map', new google.maps.Map(canvas, options));
-    }
-    this.setZoom();
-    this.setCenter();
-    if (this.get('shouldFit')) {
-      this.fitToMarkers();
-    }
+    this._super(...arguments);
+    Ember.run.next(() => {
```

Comment:
Hi @jeffrey008, the warning/deprecation says that a _certain property_ ... 
```
 was modified inside the didInsertElement hook. You should never change properties on components, services or models during didInsertElement because it causes significant performance degradation.
```
So, one simple way to solve this is by moving the property set part to the very _next_ run loop, different than the current one where the `didInsertElement` takes place. This is a well known design pattern when you need to mutate local properties on this particular hook that will avoid recursive re-renders, thus preventing significant performance lost.